osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs

```diff
@@ -43,6 +44,26 @@ public CarouselBeatmapSet(BeatmapSetInfo beatmapSet)
                       .ForEach(AddChild);
         }
 
+        public float YPosition;
+
+        public void UpdateYPosition(double elapsed)
```

Comment:
models storing drawable specific values is a huge regression. this shouldn't need to exist.

Reply:
I don't think there's any way to both have drawable pooling and have all this logic still be calculated. If the drawable panel is off screen and released back into pool, there just isn't any other place that could calculate the Y position. 

I would argue this logic has nothing to do with "drawable specific values". If ``YPosition`` sounds too related to drawables, renaming it to ``ScrollPosition`` might be less related.

Reply:
It's data directly used to drawing on screen, therefore it's drawable specific.

An alternative that wouldn't be drawable specific would be some sort of index on the carousel.

Reply:
Like move that logic to ``CarouselRoot : CarouselGroupEagerSelect`` ?? Calculate it all at the same time?

Reply:
Maybe, not necessarily, not sure. A viewmodel-like structure wrapping the model might be an answer. You'd have to try and see how you go.

Reply:
At *very least*, such a value can not be updated every frame as you have changed it to be. This is the incorrect approach at this whole thing.